### PR TITLE
support for Mastodon translation functions

### LINKED ISSUE #245: Post.application init param doesn't match the property
The Post application property is optional so the corresponding init param should be changed from

```swift
application: TootApplication,
```
to
```swift
application: TootApplication?,
```

### PR DESCRIPTION
**Pull request by technicat**
Access to the translation functions added in Mastodon 4.x via TootClient.getPostTranslation and TootClient.getTranslationLanguages. Includes fix #245 to facilitate Post copy (I place the translation results in a copy of the original post, the Post copy init is in a TootSDK extension package at https://github.com/technicat/fedicat)

### CODE REVIEW

File: Sources/TootSDK/Models/PostTranslationParams.swift [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+//
+//  Created by Philip Chu on 1/30/24.
+//
+
+import Foundation
+
+/// Params to translate a post
+public struct PostTranslationParams: Codable {
+    /// (ISO 639 language code).
+    /// The status content will be translated into this language.
```
Comment (kkostov) [inline Sources/TootSDK/Models/PostTranslationParams.swift:13]:
```suggestion
    /// The post content will be translated into this language.
```

File: Sources/TootSDK/Models/Translation.swift [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    /// The language of the source text, as auto-detected by the machine translation provider.
+    /// ISO 639 language code
+    public var detectedSourceLanguage: String
+    /// The service that provided the machine translation.
+    public var provider: String
+    /// The target language
+    /// Not in the Mastodon spec, but in the Mastodon code
+    public var language: String
+
+    public struct MediaAttachment: Codable, Hashable, Identifiable, Sendable {
```
Comment (kkostov) [inline Sources/TootSDK/Models/Translation.swift:31]: I'd like the name here to better reflect that this is not an actual attachment but a translation for an existing media.

```suggestion
    public struct MediaAttachmentTranslation: Codable, Hashable, Identifiable, Sendable {
```

File: Sources/TootSDK/Models/Translation.swift [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            self.description = description
+        }
+
+        /// The ID of the attachment in the database.
+        public var id: String
+        /// Alternate text that describes what is in the media attachment, to be used for the visually impaired or when media attachments do not load.
+        public var description: String?
+    }
+
+    public struct Poll: Codable, Hashable, Identifiable, Sendable {
```
Comment (kkostov) [inline Sources/TootSDK/Models/Translation.swift:47]:
```suggestion
    public struct PollTranslation: Codable, Hashable, Identifiable, Sendable {
```

File: Sources/TootSDK/TootClient/TootClient+Post.swift [hunk 4]
```diff
@@ -242,6 +242,20 @@ extension TootClient {
 
 }
 
+extension TootClient {
+
+    /// Translate the post content into some language.
+    public func getPostTranslation(id: String, params: PostTranslationParams? = nil) async throws -> Translation {
```
Comment (kkostov) [inline Sources/TootSDK/TootClient/TootClient+Post.swift:248]:
```suggestion
    public fund publishPostTranslation(id: String, params: PostTranslationParams? = nil) async throws -> Translation {
```

Comment (technicat) [reply Sources/TootSDK/TootClient/TootClient+Post.swift:248]: I can commit this but I believe it's wrong, you're not publishing a translation you're querying a translation.

File: Sources/TootSDK/Models/Translation.swift [hunk 5]
```diff
@@ -0,0 +1,70 @@
+//
+//  Translation.swift
+//
+//
+//  Created by Philip Chu on 1/30/24.
+//
+
+import Foundation
+
+public struct Translation: Codable {
```
Comment (kkostov) [inline Sources/TootSDK/Models/Translation.swift:10]: I think none of the translation attributes are guaranteed to be provided. The fields should be optional.

File: Sources/TootSDK/TootClient/TootClient+Post.swift (hunk 4 above)
Comment (kkostov) [reply Sources/TootSDK/TootClient/TootClient+Post.swift:248]: Oh you're right of course!

**Comment by kkostov** [comment]
Thanks a lot @technicat. I noted a few minor tweaks, let me know if you will get a chance to apply them or I could do so on my side. Also thanks for including a fix for #245 ! 🎉

**Comment by technicat** [comment]
I'll hold off on more changes (don't agree with the get->publish renaming, PollTranslation makes more sense after it's detached otherwise we have Translation.PollTranslation, and more optionals might be a good idea in Translation but for now it matches the spec (which actuallly is pretty buggy) and all the test cases I've tried and so far Mastodon is the only implementation. You can change it how you like, of course.

**Comment by kkostov** [comment]
Awesome @technicat, looks perfect to me :) 
I do agree with your comments for the naming - all good suggestions.

**Review by kkostov — APPROVED** [review]

**Pull request merged**